<?xml version="1.0"?>

<launch>

    <!-- ARGS -->
    <arg name="bagfile" default="/home/nvidia/seagate/bagfiles/2018-11-11/belltower_walk.bag" />
    <arg name="neural_net_graph_path" default="$(find segmentation_projection)/config/graph_frozen_nclt_400p.pb" />	
    
    <!-- BAGFILE -->
    <node pkg="rosbag" type="play" name="player" output="screen" args="$(arg bagfile) -q -l -r 0.2" />

    <!-- RVIZ -->
    <?ignore
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(arg rvizfile)"/>
    ?>

    <!-- Run segmentation node -->

    <node pkg="segmentation_projection"     type="segmentation_projection_node" name="segmentation_projection_node" output="screen" clear_params="true" required="true" >
      
      <param name="labeled_pointcloud"   type="string" value="labeled_pointcloud" />
    
      <param name="neural_net_graph"     type="string" value="$(arg neural_net_graph_path)"/>
      <param name="is_output_distribution" type="int" value="0" />
      <param name="neural_net_input_width" type="int"  value="800" />
      <param name="neural_net_input_height" type="int" value="400" />
    
      <param name="lidar"                type="string" value="/velodyne_points" />
    
      <param name="camera_num"           type="int"    value='1' />
      <param name="image_0"              type="string" value="/camera1/color/image_raw" />
      <param name="cam_intrinsic_0"    type="string" value="$(find segmentation_projection)/config/realsense_intrinsic.npy"/>	
      <param name="cam2lidar_file_0"     type="string" value="$(find segmentation_projection)/config/realsense2lidar.npy"/>
    
  </node>




 
</launch>
